Solution1/Solution1.Tests/ConsoleApp/Services/PerformerCommandsServiceTests.cs

```diff
@@ -0,0 +1,115 @@
+            _weatherServiceMock = new Mock<IWeatherServiсe>();
+            _performerCommandsService = new PerformerCommandsService(_loggerMock.Object, _weatherServiceMock.Object);
+        }
+
+        [Fact]
+        public async Task GetCurrentWeatherAsync_EnterCityName_ShowStringRepresentationAsync()
+        {
+            // Arrange
+            var cityName = "Minsk";
+            var weatherDto = new WeatherDTO() { CityName = cityName, Temp = 10, Comment = "It's fresh." };
+            _weatherServiceMock
+                .Setup(weatherApiService =>
+                    weatherApiService
+                    .GetByCityNameAsync(cityName))
+                .ReturnsAsync(weatherDto);
+
+            var consoleOutput = new StringWriter();
+            Console.SetOut(consoleOutput);
+
+            Console.SetIn(new StringReader(string.Format("Minsk{0}", Environment.NewLine)));
+
+            //Act
+            await _performerCommandsService.GetCurrentWeatherAsync();
+
+            //Assert
+            var expected = string.Format("Please, enter city name:{0}In Minsk 10 C. It's fresh.{0}", Environment.NewLine);
+            Assert.Equal(expected, consoleOutput.ToString());
+        }
+
+        [Fact]
+        public async Task GetForecastByCityNameAsync_EnterCityNameAndCountDay_ShowMultiStringRepresentationAsync()
+        {
+            // Arrange
+            var cityName = "Minsk";
+            var countDay = 3;
+
+            var weatherForPeriod = new List<WeatherForDateDTO>()
+            {
+                new WeatherForDateDTO() { DateTime = new DateTime(2022, 12, 10), Temp = -1, Comment = "Dress warmly." },
+                new WeatherForDateDTO() { DateTime = new DateTime(2022, 12, 11), Temp = 10, Comment = "It's fresh."  },
+                new WeatherForDateDTO() { DateTime = new DateTime(2022, 12, 12), Temp = 25, Comment = "Good weather."  },
+                new WeatherForDateDTO() { DateTime = new DateTime(2022, 12, 13), Temp = 35, Comment = "It's time to go to the beach."  }
+            };
+
+            var forecastWeatherDTO = new ForecastWeatherDTO() { CityName = cityName, WeatherForPeriod = weatherForPeriod };
+
+            _weatherServiceMock
+                .Setup(weatherApiService =>
+                    weatherApiService.GetForecastByCityNameAsync(cityName, countDay)
+                    )
+                .ReturnsAsync(forecastWeatherDTO);
+
+            var consoleOutput = new StringWriter();
+            Console.SetOut(consoleOutput);
+            Console.SetIn(new StringReader(string.Format("{0}{1}{2}{1}", cityName, Environment.NewLine, countDay)));
+
+            //Act
+            await _performerCommandsService.GetForecastByCityNameAsync();
+
+            //Assert
+            var expected = "Please, enter city name:\r\n" +
+                "Please, enter count day:\r\n" +
+                "Minsk weather forecast: \n" +
+                "Day 0 (10 декабря 2022 г.): -1,0 C. Dress warmly. \n" +
```

Comment:
it won't work for you future collogues from USA

Reply:
+